Stripe/STPPaymentCardTextFieldViewModel.m

```diff
@@ -106,40 +100,42 @@ - (STPCardBrand)brand {
     return [STPCardValidator brandForNumber:self.cardNumber];
 }
 
-- (STPCardValidationState)validationStateForField:(STPCardFieldType)fieldType {
-    switch (fieldType) {
-        case STPCardFieldTypeNumber:
-            return [STPCardValidator validationStateForNumber:self.cardNumber validatingCardBrand:YES];
-            break;
-        case STPCardFieldTypeExpiration: {
-            STPCardValidationState monthState = [STPCardValidator validationStateForExpirationMonth:self.expirationMonth];
-            STPCardValidationState yearState = [STPCardValidator validationStateForExpirationYear:self.expirationYear inMonth:self.expirationMonth];
-            if (monthState == STPCardValidationStateValid && yearState == STPCardValidationStateValid) {
-                return STPCardValidationStateValid;
-            } else if (monthState == STPCardValidationStateInvalid || yearState == STPCardValidationStateInvalid) {
-                return STPCardValidationStateInvalid;
-            } else {
-                return STPCardValidationStateIncomplete;
-            }
-            break;
-        }
-        case STPCardFieldTypeCVC:
-            return [STPCardValidator validationStateForCVC:self.cvc cardBrand:self.brand];
-        case STPCardFieldTypePostalCode:
-            if (self.postalCode.length > 0) {
-                return STPCardValidationStateValid;
-            } else {
-                return STPCardValidationStateIncomplete;
-            }
+- (STPCardValidationState)validationStateForCVC {
+    return [STPCardValidator validationStateForCVC:self.cvc cardBrand:self.brand];
+}
+
+- (STPCardValidationState)validationStateForExpiration {
+    STPCardValidationState monthState = [STPCardValidator validationStateForExpirationMonth:self.expirationMonth];
+               STPCardValidationState yearState = [STPCardValidator validationStateForExpirationYear:self.expirationYear inMonth:self.expirationMonth];
+               if (monthState == STPCardValidationStateValid && yearState == STPCardValidationStateValid) {
+                   return STPCardValidationStateValid;
+               } else if (monthState == STPCardValidationStateInvalid || yearState == STPCardValidationStateInvalid) {
+                   return STPCardValidationStateInvalid;
+               } else {
+                   return STPCardValidationStateIncomplete;
+               }
+}
+
+- (STPCardValidationState)validationStateForPostalCode {
+    if (self.postalCode.length > 0) {
+        return STPCardValidationStateValid;
+    } else {
+        return STPCardValidationStateIncomplete;
     }
 }
 
+- (void)validationStateForCardNumberWithHandler:(void (^)(STPCardValidationState))handler {
+    [STPBINRange retrieveBINRangesForPrefix:self.cardNumber completion:^(__unused NSArray<STPBINRange *> * _Nullable ranges, __unused NSError * _Nullable error) {
+        handler([STPCardValidator validationStateForNumber:self.cardNumber validatingCardBrand:YES]);
```

Comment:
I think I slightly prefer `retrieveBINRanges` because the expectation is that you always get the BIN ranges with this method even if it doesn't have to make a network request to update them, but don't feel strongly.

I could see it on `STPCardValidator`, but I'd want to track the retrieved ranges in the same place as the hard-coded ranges so `STPBINRange` seemed more sensible. Open to ideas for that though